### ISSUE TITLE
Cannot handle env COMPILER_PATH on RHEL-based system 

### ISSUE THREAD

**Issue by ZQyou** [apptainer/singularity#4660]
### Version of Singularity
With Vagrant box singularity-3.4-centos-7-64
```
$ singularity version
3.4.1-1
```

### Expected behavior
```
$ export COMPILER_PATH=1234
$ export SINGULARITYENV_COMPILER_PATH=5678
$ singularity shell library://library/default/alpine
Singularity> echo $COMPILER_PATH
5678
```

### Actual behavior
```
$ export COMPILER_PATH=1234
$ export SINGULARITYENV_COMPILER_PATH=5678
$ singularity shell library://library/default/alpine
Singularity> echo $COMPILER_PATH
1234
```

### Steps to reproduce this behavior
It can be reproduced on my other RHEL system but not on Ubuntu system (tested on Vagrant box singularity-3.4-ubuntu-bionic64)

### What OS/distro are you running
```
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
[ .. more .. ]
```

### How did you install Singularity
Vagrant box

**Comment by dtrudg** [comment]
Please could you provide the `singularity version` output from your ubuntu vagrant box that is working? There have now been 3 releases of 3.4, with 3.4.2 being the latest version and I'd like to be able to rule out a difference on version rather than the OS.

**Comment by ZQyou** [comment]
Singularity version on my ubuntu vagrant box: 
```
$ singularity version
3.4.1-1
```

**Comment by dtrudg** [comment]
I don't see this issue with `3.5.2`

```
dtrudg@fedora-dev:~
09:16 AM $ export COMPILER_PATH=1234
dtrudg@fedora-dev:~
09:27 AM $ export SINGULARITYENV_COMPILER_PATH=5678
dtrudg@fedora-dev:~
09:27 AM $ singularity shell library://library/default/alpine
INFO:    Downloading library image
Singularity> echo $COMPILER_PATH
5678
```

Please could you report here if you still see it with the latest version of Singularity?

**Comment by ZQyou** [comment]
I have upgraded singularity to `3.5.2-1.el7` on our RHEL system, but the issue remains unresolved:
```
[zyou@o0550 ~]$ ml purge
[zyou@o0550 ~]$ singularity version
3.5.2-1.el7
[zyou@o0550 ~]$ export COMPILER_PATH=1234
[zyou@o0550 ~]$ export SINGULARITYENV_COMPILER_PATH=5678
[zyou@o0550 ~]$ singularity shell library://library/default/alpine
Singularity> echo $COMPILER_PATH
1234
[zyou@o0550 ~]$ env |grep COMPILER_PATH
SINGULARITYENV_COMPILER_PATH=5678
COMPILER_PATH=1234
```

**Comment by dtrudg** [comment]
There's something very strange going on here, as I'm unable to replicate from a source build on a RHEL7 VM.

Could you try building singularity from source (not the rpm), and see if it reproduces?

**Comment by cclerget** [comment]
This issue is fixed in master and will be available in next coming release 3.6 (around May) where `SINGULARITYENV_COMPILER_PATH` take precedence over `COMPILER_PATH`.

Closing it as fixed. Feel free to re-open it if needed
